dolphinscheduler-storage-plugin/dolphinscheduler-storage-cos/src/main/java/org/apache/dolphinscheduler/plugin/storage/cos/CosStorageOperator.java

```diff
@@ -0,0 +1,335 @@
+            return bufferedReader
+                    .lines()
+                    .skip(skipLineNums)
+                    .limit(limit)
+                    .collect(Collectors.toList());
+        }
+    }
+
+    @Override
+    public List<StorageEntity> listStorageEntity(String resourceAbsolutePath) {
+        resourceAbsolutePath = transformCOSKeyToAbsolutePath(resourceAbsolutePath);
+
+        ListObjectsRequest request = new ListObjectsRequest();
+        request.setBucketName(bucketName);
+        request.setPrefix(resourceAbsolutePath);
+        request.setDelimiter("/");
+
+        ObjectListing result = cosClient.listObjects(request);
+
+        return result.getObjectSummaries()
+                .stream()
+                .map((COSObjectSummary summary) -> {
+                    ObjectMetadata metadata = new ObjectMetadata();
+                    metadata.setContentLength(summary.getSize());
+                    metadata.setLastModified(summary.getLastModified());
+                    COSObject object = new COSObject();
+                    object.setObjectMetadata(metadata);
+                    object.setKey(summary.getKey());
+                    return transformCOSObjectToStorageEntity(object);
+                })
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<StorageEntity> listFileStorageEntityRecursively(String resourceAbsolutePath) {
+        resourceAbsolutePath = transformCOSKeyToAbsolutePath(resourceAbsolutePath);
+
+        Set<String> visited = new HashSet<>();
+        List<StorageEntity> storageEntityList = new ArrayList<>();
+        LinkedList<String> foldersToFetch = new LinkedList<>();
+        foldersToFetch.addLast(resourceAbsolutePath);
+
+        while (!foldersToFetch.isEmpty()) {
+            String pathToExplore = foldersToFetch.pop();
+            visited.add(pathToExplore);
+            List<StorageEntity> storageEntities = listStorageEntity(pathToExplore);
+            for (StorageEntity entity : storageEntities) {
+                if (entity.isDirectory()) {
+                    if (visited.contains(entity.getFullName())) {
+                        continue;
+                    }
+                    foldersToFetch.add(entity.getFullName());
+                }
+            }
+            storageEntityList.addAll(storageEntities);
+        }
+
+        return storageEntityList;
+    }
+
+    @Override
+    public StorageEntity getStorageEntity(String resourceAbsolutePath) {
+        String cosKey = transformCOSKeyToAbsolutePath(resourceAbsolutePath);
+
+        COSObject object = cosClient.getObject(bucketName, cosKey);
+        return transformCOSObjectToStorageEntity(object);
+    }
+
+    public void ensureBucketSuccessfullyCreated(String bucketName) {
+        if (StringUtils.isBlank(bucketName)) {
+            throw new IllegalArgumentException("resource.tencent.cloud.cos.bucket.name is empty");
```

Comment:
It's better to use `Constants` to replace `resource.tencent.cloud.cos.bucket.name`